### PR TITLE
foam.net.RetryHTTPRequest: Retry requests that reject-on-send()

### PR DESCRIPTION
**Pull request by mdittmer**

### CODE REVIEW

File: src/foam/net/RetryHTTPRequest.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  listeners: [
+    function onError(error) {
+      this.currentTry_++;
+      this.warn('RetryHTTPRequest: Try #' + this.currentTry_ +
+                ' failed on ' + error);
+      if ( this.currentTry_ < this.numTries ) {
+        return this.send();
+      } else {
+        this.error('RetryHTTPRequest: Max tries reached');
+        throw new Error('RetryHTTPRequest: Max tries reached');
```
Comment (jacksonic) [inline src/foam/net/RetryHTTPRequest.js:65]: Should the underlying error also be included?

Comment (mdittmer) [reply src/foam/net/RetryHTTPRequest.js:65]: Append it to the message? AFAICT JS errors don't have a great model for this.

Comment (mdittmer) [reply src/foam/net/RetryHTTPRequest.js:65]: Also, I'm not accumulating the errors on each attempt, so the last error may prove to be a red herring...

Comment (mdittmer) [reply src/foam/net/RetryHTTPRequest.js:65]: @jacksonic LMK if feels strongly we should fix this in a follow-up PR. I'm going to land this with some other changes that I want to merge cleanly into beta-1.

Comment (jacksonic) [reply src/foam/net/RetryHTTPRequest.js:65]: True.

**Review by jacksonic — APPROVED** [review]

**Pull request merged**